### DISCUSSION TITLE
Swup stuck when no transition found

### DISCUSSION THREAD

**Discussion by CHolitas** [swup/swup#399] [Help] [answered]
Hello I have a problem, where Swup gets stuck in `is-changing is-leaving is-animating`
Can someone help me?

**Comment by AugustMiller** [comment]
Can you make sure you don't have any classes with the `transition-` prefix in your document? This is the most common reason for Swup failing to move to the next page, as it waits for these elements to all produce a transition-end event in JS.

**Reply by CHolitas** [reply]
Hello, no the only one I have with `transition-` is ` <main id="swup" class="transition-fade">`

**Reply by AugustMiller** [reply]
…and does that have any CSS transitions applied to it?

**Reply by CHolitas** [reply]
Yes, off course.

**Reply by AugustMiller** [reply]
Huh! Ok—if you could follow up with another comment and share the smallest possible reproducible example, that'd be useful. Remove all but the Swup initializer, minimal styles, and the two HTML pages you're having trouble with, and something might jump out!

**Reply by CHolitas** [reply]
The thing is that when I click to the second page swipe adds the `is-changing is-leaving is-animating` and it gets sucked there so the other page doesn't load, the thing is that the url changes so for example I go from `index.html` to `second.html` the url changes to `localhost/second.html` and it gets sucked there

PS: I also use scrollreveal.js on my project, do you believe it has something to do?

**Comment by lightyearsagency** [comment]
I'm having a similar issue here
https://github.com/swup/swup/discussions/462

But it's related to navigating away from a page that would contain the normal header\footer (It's a media file), and then clicking the 'back' button.

Are we both perhaps encountering something where SWUP doesn't recognise that it's been navigated to?

**Comment by daun** [comment, marked as answer]
Swup 2.0.17 will skip the transition and emit a warning if no transition was found, instead of getting stuck. This should make debugging this type of issue easier as well.

**Reply by lightyearsagency** [reply]
Thanks for that!
It's now throwing errors instead of getting stuck.
